r/src/expression.cpp

```diff
@@ -19,93 +19,98 @@
 
 #if defined(ARROW_R_WITH_ARROW)
 
+#include <arrow/compute/api_scalar.h>
 #include <arrow/dataset/api.h>
 namespace ds = ::arrow::dataset;
 
+std::shared_ptr<ds::Expression> Share(ds::Expression expr) {
+  return std::make_shared<ds::Expression>(std::move(expr));
+}
+
```

Comment:
Would it be appropriate in this PR to add the `compute::CallFunction` equivalent for dataset expressions? Before I can add any new expressions, we'll need it. 

I'm guessing it looks like:

```
std::shared_ptr<ds::Expression> dataset___expr(std::string func_name, cpp11::list args, cpp11::list options)
```

and calls `ds::call()` inside it?

Reply:
I'll add that, thanks